dist_run.py

```diff
@@ -4,32 +4,65 @@
 # This source code is licensed under the license found in the
 # LICENSE file in the root directory of this source tree.
 
+# Run command:
+# torchrun --nproc-per-node 4 dist_run.py
+
 import torch
 import torch.distributed as dist
+from torch.distributed.pipelining import PipelineStage, ScheduleGPipe
 
 from build.model import TransformerArgs
-from build.model_dist import Transformer
+from build.model_dist import TransformerStage
 
 # Model config
 config = TransformerArgs.from_name("Transformer-2-7b-chat-hf")
 print(config)
 
 # Construct a device mesh with available devices (multi-host or single host)
-device_mesh = dist.init_device_mesh("cuda", (2,), mesh_dim_names=("tp",))
+device_mesh = dist.init_device_mesh("cuda", (2, 2), mesh_dim_names=("pp", "tp"))
```

Comment:
if this file expands in future, maybe better to functionalize these different stages ala 
create_device_mesh(mesh_shape=(2,2))
setup_model
create_pipeline_stage
etc. 
It's clear atm what's happening but if this file becomes larger/more dynamic then maybe it would make sense in the future.

Reply:
Sure makes sense.